libs/eavmlib/src/gpio.erl

```diff
@@ -49,13 +57,27 @@
 ]).
 
 -type gpio() :: pid().
--type pin() :: non_neg_integer() | {atom(), non_neg_integer()}.
--type direction() :: input | output | output_od.
+%% This is the pid returned by `gpio:start/0'.
+-type pin() :: non_neg_integer() | stm32_pin().
+%% The pin definition on the STM32 platform varies from ESP32 and PR2040.
+-type stm32_pin() :: {gpio_bank(), 0..15}.
```

Comment:
Maybe call this `banked_pin` so it is not stm-specific?

Reply:
The problem is that architecture is stm32 specific, and I want to make it very clear that that particular format only works on stm32. With a generic name it might mislead users into attempting to "bank" together pins on other platforms. Inversely I did not want stm32 users to be frustrated because the correct pin format is not obvious.